File: Atcoder_#342/ruby/b_which_is_ahead.rb
n = gets.to_i
p = gets.chomp.split.map(&:to_i)
q = gets.to_i

positions = {}
p.each_with_index do |person, index|
  positions[person] = index
end

q.times do
  Ai, Bi = gets.chomp.split.map(&:to_i)
  if positions[Ai] < positions[Bi]
    puts Ai
  else
    puts Bi
  end
end

File: Atcoder_#342/javascript/b_which_is_ahead.js
const fs = require('fs');

function main(input) {
  const lines = input.trim().split('\n');
  
  const n = parseInt(lines[0], 10);
  
  const p = lines[1].split(' ').map(Number);
  
  const q = parseInt(lines[2], 10);

  const positions = {};
  p.forEach((person, index) => {
    positions[person] = index;
  });

  for (let i = 3; i < 3 + q; i++) {
    const [Ai, Bi] = lines[i].split(' ').map(Number);
    
    if (positions[Ai] < positions[Bi]) {
      console.log(Ai);
    } else {
      console.log(Bi);
    }
  }
}

main(fs.readFileSync('/dev/stdin', 'utf8'));
